# HG changeset patch
# User Rik <rik@octave.org>
# Date 1453044175 28800
#      Sun Jan 17 07:22:55 2016 -0800
# Node ID 0ebb8e26ebbc7f7c19ed97dd3f513a431bc3b3f4
# Parent  429f8390340076c7f4a4ba1417c8e00aaa12bb4b
build: Document Automake failure to 'include scripts/@ftp/module.mk'.

* scripts/@ftp/module.mk: Add explanation why this file is empty.

* scripts/module.mk: Delimit included scripts/@ftp/module.mk file with
line of '#'.

diff --git a/scripts/@ftp/module.mk b/scripts/@ftp/module.mk
--- a/scripts/@ftp/module.mk
+++ b/scripts/@ftp/module.mk
@@ -1,5 +1,10 @@
-## FIXME: including scripts/@ftp/module.mk fails.
-## Is that an automake bug?
-## Would getting the quoting character right fix it?
+## Automake fails to process "include scripts/@ftp/module.mk" in the directory
+## above.  All of the commands which would normally be in this file were
+## manually placed in scripts/module.mk to avoid using the "include" directive.
 ##
-## What should be in this file is in scripts/module.mk.
+## This is an Automake bug.  Automake has switched to a Perl backend which uses
+## the following pattern to detect a path:
+##
+## my $PATH_PATTERN = '(\w|[+/.-])+';
+##
+## This pattern only includes alphanumeric, '_', and [+/.-], but not "@".
diff --git a/scripts/module.mk b/scripts/module.mk
--- a/scripts/module.mk
+++ b/scripts/module.mk
@@ -1,45 +1,14 @@
 scripts_EXTRA_DIST =
 
 scripts_CLEANFILES =
 scripts_DISTCLEANFILES =
 scripts_MAINTAINERCLEANFILES =
 
-## FIXME: including scripts/@ftp/module.mk fails.  Is that an automake bug?
-
-FCN_FILE_DIRS += scripts/@ftp
-
-scripts_@ftp_FCN_FILES = \
-  scripts/@ftp/ascii.m \
-  scripts/@ftp/binary.m  \
-  scripts/@ftp/cd.m  \
-  scripts/@ftp/close.m  \
-  scripts/@ftp/delete.m  \
-  scripts/@ftp/dir.m  \
-  scripts/@ftp/display.m  \
-  scripts/@ftp/ftp.m  \
-  scripts/@ftp/loadobj.m  \
-  scripts/@ftp/mget.m  \
-  scripts/@ftp/mkdir.m  \
-  scripts/@ftp/mput.m  \
-  scripts/@ftp/rename.m  \
-  scripts/@ftp/rmdir.m  \
-  scripts/@ftp/saveobj.m
-
-scripts_@ftpdir = $(fcnfiledir)/@ftp
-
-scripts_@ftp_DATA = $(scripts_@ftp_FCN_FILES)
-
-FCN_FILES += $(scripts_@ftp_FCN_FILES)
-
-PKG_ADD_FILES += scripts/@ftp/PKG_ADD
-
-DIRSTAMP_FILES += scripts/@ftp/$(octave_dirstamp)
-
 include scripts/audio/module.mk
 include scripts/deprecated/module.mk
 include scripts/elfun/module.mk
 include scripts/general/module.mk
 include scripts/geometry/module.mk
 include scripts/gui/module.mk
 include scripts/help/module.mk
 include scripts/image/module.mk
@@ -65,16 +34,51 @@ include scripts/startup/module.mk
 include scripts/statistics/base/module.mk
 include scripts/statistics/distributions/module.mk
 include scripts/statistics/models/module.mk
 include scripts/statistics/tests/module.mk
 include scripts/strings/module.mk
 include scripts/testfun/module.mk
 include scripts/time/module.mk
 
+## include scripts/@ftp/module.mk
+## The include above fails because Automake cannot process the '@' character.
+## As a work around, the contents of scripts/@ftp/module.mk are placed directly
+## in this module.mk file.
+######################## include scripts/@ftp/module.mk ########################
+FCN_FILE_DIRS += scripts/@ftp
+
+scripts_@ftp_FCN_FILES = \
+  scripts/@ftp/ascii.m \
+  scripts/@ftp/binary.m  \
+  scripts/@ftp/cd.m  \
+  scripts/@ftp/close.m  \
+  scripts/@ftp/delete.m  \
+  scripts/@ftp/dir.m  \
+  scripts/@ftp/display.m  \
+  scripts/@ftp/ftp.m  \
+  scripts/@ftp/loadobj.m  \
+  scripts/@ftp/mget.m  \
+  scripts/@ftp/mkdir.m  \
+  scripts/@ftp/mput.m  \
+  scripts/@ftp/rename.m  \
+  scripts/@ftp/rmdir.m  \
+  scripts/@ftp/saveobj.m
+
+scripts_@ftpdir = $(fcnfiledir)/@ftp
+
+scripts_@ftp_DATA = $(scripts_@ftp_FCN_FILES)
+
+FCN_FILES += $(scripts_@ftp_FCN_FILES)
+
+PKG_ADD_FILES += scripts/@ftp/PKG_ADD
+
+DIRSTAMP_FILES += scripts/@ftp/$(octave_dirstamp)
+####################### end include scripts/@ftp/module.mk #####################
+
 image_DATA += $(SCRIPTS_IMAGES)
 
 GEN_FCN_FILES_IN = $(GEN_FCN_FILES:.m=.in)
 
 ALL_LOCAL_TARGETS += \
   $(GEN_FCN_FILES) \
   $(PKG_ADD_FILES) \
   $(JAR_FILES)
